### ISSUE TITLE
Manage cluster metadata with RAFT for better reliability

### ISSUE THREAD

**Issue by etiennedi** [weaviate/weaviate#4280]
## tl;dr
The outcome of #3515. All cluster meta data including schema information will be managed with RAFT in the future. This increases fault-tolerance and fixes "schema out of sync" errors once and for all.

## Issues addressed by RAFT

* `schema out of sync` startup errors: RAFT will keep the schema in sync.
* `concurrent transaction` error on schema change: The current schema tx cannot handle concurrency, the RAFT setup will.

## Migration Journey
The migration journey will be designed in a way that no extra steps are necessary. A regular rolling update will handle the migration automatically.

## Downgrading after upgrading
After upgrading to the RAFT-based version (likely v1.25), downgrading directly will no longer be supported. For those wanting to go back to an older version, a backup-and-restore is required.

**Comment by dirkkul** [comment]
also done I think
